src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStepExecution.java

```diff
@@ -45,24 +50,28 @@
 
     @Override
     public boolean start() throws Exception {
-
         Cloud cloud = Jenkins.getInstance().getCloud(cloudName);
         if (cloud == null) {
             throw new AbortException(String.format("Cloud does not exist: %s", cloudName));
         }
         if (!(cloud instanceof KubernetesCloud)) {
-            throw new AbortException(String.format("Cloud is not a Kubernetes cloud: %s (%s)", cloudName,
-                    cloud.getClass().getName()));
+            throw new AbortException(
+                    String.format("Cloud is not a Kubernetes cloud: %s (%s)", cloudName, cloud.getClass().getName()));
         }
         KubernetesCloud kubernetesCloud = (KubernetesCloud) cloud;
 
         Run<?, ?> run = getContext().get(Run.class);
+        if (kubernetesCloud.isUsageRestricted()) {
```

Comment:
Not the ideal design (ideally `PodTemplateStepExecution` should not depend on `KubernetesFolderProperty`) but could be revisited later by introducing a new extension point.

Reply:
no idea how to do this and, to be totally honest, I don't feel the urge to dig any deeper into the internals of Jenkins and its quirks.